### PR TITLE
feat: resolve spotbugs and scalastyle issues

### PR DESCRIPTION
**Pull request by tobegit3hub**
* Set `revision` instead custom property as version so that maven will print warning info
* Upgrade `maven-surefire-plugin` to `2.21.0` to remove the useless warning info
* Remove the duplicated scala compile stage which is meaningless and take much more time
* Ignore the useless warning info for shade phase
* Run spotbugs when compiling and fail for `HIGH` warning
* Add the spotbugs configuration file to ignore checking for generated classes
* Update the scalastyle and the code the fix all the style warning
* Use `StreamReader` and `StreamWriter` to avoid using strings without charset which may print warning info
* Fix java class comment to avoid warning info from java docs plugin
* Use new `java.sql.Date` constructor instead of the deprecated one to avoid warning info
* Turn off the scalastyle for `import spark.implicits._`

### CODE REVIEW

**Comment by tobegit3hub** [comment]
Now the compile message is clean and check code style for every commit.

Screen Shot 2021-07-28 at 6 37 19 PM

File: java/openmldb-batch/scala_style.xml [hunk 1]
```diff
@@ -100,7 +105,8 @@
             <parameter name="doubleLineAllowed"><![CDATA[false]]></parameter>
         </parameters>
     </check>
-    <check level="warning" class="org.scalastyle.scalariform.MethodLengthChecker" enabled="true">
+    <!-- Update by 4pardigm -->
```
Comment (zekai427) [inline java/openmldb-batch/scala_style.xml:108]: add init status

File: java/openmldb-batch/scala_style.xml [hunk 2]
```diff
@@ -87,7 +91,8 @@
             <parameter name="maxTypes"><![CDATA[30]]></parameter>
         </parameters>
     </check>
-    <check level="warning" class="org.scalastyle.scalariform.CyclomaticComplexityChecker" enabled="true">
+    <!-- Update by 4paradigm -->
```
Comment (zekai427) [inline java/openmldb-batch/scala_style.xml:94]: add init status

File: java/openmldb-batch/scala_style.xml [hunk 3]
```diff
@@ -69,8 +71,10 @@
     </check>
     <check level="warning" class="org.scalastyle.scalariform.NoWhitespaceBeforeLeftBracketChecker" enabled="true"/>
     <check level="warning" class="org.scalastyle.scalariform.NoWhitespaceAfterLeftBracketChecker" enabled="true"/>
-    <check level="warning" class="org.scalastyle.scalariform.ReturnChecker" enabled="true"/>
-    <check level="warning" class="org.scalastyle.scalariform.NullChecker" enabled="true"/>
+    <!-- Update by 4paradigm -->
+    <check level="warning" class="org.scalastyle.scalariform.ReturnChecker" enabled="false"/>
+    <!-- Update by 4paradigm -->
```
Comment (zekai427) [inline java/openmldb-batch/scala_style.xml:76]: add init status

File: java/openmldb-batch/scala_style.xml [hunk 4]
```diff
@@ -69,8 +71,10 @@
     </check>
     <check level="warning" class="org.scalastyle.scalariform.NoWhitespaceBeforeLeftBracketChecker" enabled="true"/>
     <check level="warning" class="org.scalastyle.scalariform.NoWhitespaceAfterLeftBracketChecker" enabled="true"/>
-    <check level="warning" class="org.scalastyle.scalariform.ReturnChecker" enabled="true"/>
-    <check level="warning" class="org.scalastyle.scalariform.NullChecker" enabled="true"/>
+    <!-- Update by 4paradigm -->
```
Comment (zekai427) [inline java/openmldb-batch/scala_style.xml:74]: add init status

**Review by zekai427 — APPROVED** [review]
LGTM

**Review by imotai — APPROVED** [review]

**Pull request merged**